tests/test_register.py

```diff
@@ -40,3 +45,37 @@ def test_sydent_rejects_invalid_hostname(self):
         request.render(self.sydent.servlets.registerServlet)
 
         self.assertEqual(channel.code, 400)
+
+    @parameterized.expand(
+        [
+            (twisted.internet.error.DNSLookupError(),),
+            (twisted.internet.error.TimeoutError(),),
+            (twisted.internet.error.ConnectionRefusedError(),),
+            # Naughty: strictly we're supposed to initialise a ResponseNeverReceived
+            # with a list of 1 or more failures.
+            (twisted.web.client.ResponseNeverReceived([]),),
+        ]
+    )
+    def test_connection_failure(self, exc: Exception) -> None:
+        self.sydent.run()
+        request, channel = make_request(
+            self.sydent.reactor,
+            "POST",
+            "/_matrix/identity/v2/account/register",
+            content={
+                "matrix_server_name": "matrix.alice.com",
+                "access_token": "back_in_wonderland",
+            },
+        )
+        servlet = self.sydent.servlets.registerServlet
+
+        def mock_get_json(*args: object, **kwargs: object) -> None:
+            raise exc
+
+        with patch.object(servlet.client, "get_json", mock_get_json):
```

Comment:
You might be able to do `side_effect=exc` and avoid `mock_get_json` existing.

Reply:
Thanks, I knew there was some incantation for this!